envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadataValidatorTest.kt

```diff
@@ -4,11 +4,35 @@ import io.envoyproxy.envoy.api.v2.DiscoveryRequest
 import io.grpc.Status
 import org.assertj.core.api.Assertions.assertThat
 import org.assertj.core.api.Assertions.assertThatExceptionOfType
+import org.junit.jupiter.api.Assertions.assertDoesNotThrow
 import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import pl.allegro.tech.servicemesh.envoycontrol.snapshot.CommunicationMode
 import pl.allegro.tech.servicemesh.envoycontrol.snapshot.OutgoingPermissionsProperties
 import pl.allegro.tech.servicemesh.envoycontrol.snapshot.SnapshotProperties
 
 class NodeMetadataValidatorTest {
+
+    companion object {
+        @JvmStatic
+        fun configurationModeNotSupported() = listOf(
+            Arguments.of(false, true, true, "ADS"),
```

Comment:
3rd argument, why boolean and not enum?

Reply:
This is configuration in properties and there we have flags.